python/pyspark/ml/pipeline.py

```diff
@@ -89,9 +87,9 @@ def setParams(self, stages=None):
         setParams(self, stages=None)
         Sets params for Pipeline.
         """
-        if stages is None:
-            stages = []
         kwargs = self.setParams._input_kwargs
+        if stages is None:
+            kwargs['stages'] = []
```

Comment:
So almost all of our things explode if you call setParams with something set to None - it gets turned into null on  the Java side and that tends to explode. We could avoid copying None params over but we might want to do that more generally than just here (and I'd probably do that in a separate PR since I'm not super sure that is the right thing to do or if we just expect people to not use None unless they really explicitly mean it).